src/LayerTree/LayerTree.tsx

```diff
@@ -75,8 +76,8 @@ export interface BaseProps {
 
 interface LayerTreeState {
   layerGroup: OlLayerGroup;
-  layerGroupRevision: null;
-  treeNodes: LayerTreeNode[];
+  layerGroupRevision?: number;
+  treeNodes: ReactElement[];
```

Comment:
This seems to be more unprecise the the previous typing.

Reply:
I agree, but somehow the typecheck was failing here. I don't exactly see why it was passing before.

Reply:
The problem is that typescript says that `return (<LayerTreeNode ... >)` is not assignable to the type `LayerTreeNode`. Do you know why this could be the case?

Reply:
Hmm could it be related to antd? And why did it work before?

Reply:
I tried to revert the types to the initial state, but the typecheck still fails. Also I reverted typescript back to `v4.0.3` which also did not help.